src/java.base/share/classes/java/net/URLClassLoader.java

```diff
@@ -488,48 +454,51 @@ public URL[] getURLs() {
-                    }                                                           //OpenJ9-shared_classes_misc
-                }                                                               //OpenJ9-shared_classes_misc
-            }                                                                   //OpenJ9-shared_classes_misc
-            ClassFinder loader = new ClassFinder(name, this);   /*ibm@80916.1*/ //OpenJ9-shared_classes_misc
-            result = AccessController.doPrivileged(loader, acc);                //OpenJ9-shared_classes_misc
-        } catch (java.security.PrivilegedActionException pae) {
-            throw (ClassNotFoundException) pae.getException();
-        }
-        if (result == null) {
-            throw new ClassNotFoundException(name);
+        /* Try to find the class from the shared cache using the class name. //OpenJ9-shared_classes_misc
+         * If we found the class and if we have its corresponding metadata  //OpenJ9-shared_classes_misc
+         * (code source and manifest entry) already cached, then we define  //OpenJ9-shared_classes_misc
+         * the class passing in these parameters.  If however, we do not    //OpenJ9-shared_classes_misc
+         * have the metadata cached, then we define the class as normal.    //OpenJ9-shared_classes_misc
+         * Also, if we do not find the class from the shared class cache,   //OpenJ9-shared_classes_misc
+         * we define the class as normal.                                   //OpenJ9-shared_classes_misc
+         */                                                                 //OpenJ9-shared_classes_misc
+        if (usingSharedClasses()) {                                         //OpenJ9-shared_classes_misc
+            SharedClassIndexHolder sharedClassIndexHolder = new SharedClassIndexHolder(); /*ibm@94142*/ //OpenJ9-shared_classes_misc
+            IntConsumer consumer = (i)->sharedClassIndexHolder.setIndex(i); //OpenJ9-shared_classes_misc
+            byte[] sharedClazz = sharedClassServiceProvider.findSharedClassURLClasspath(name, consumer); //OpenJ9-shared_classes_misc
+            if (sharedClazz != null) {                                      //OpenJ9-shared_classes_misc
+                int indexFoundData = sharedClassIndexHolder.index;          //OpenJ9-shared_classes_misc
+                SharedClassMetaData metadata = sharedClassMetaDataCache.getSharedClassMetaData(indexFoundData); //OpenJ9-shared_classes_misc
+                if (metadata != null) {                                     //OpenJ9-shared_classes_misc
+                    try {                                                   //OpenJ9-shared_classes_misc
+                        Class<?> clazz = defineClass(name, sharedClazz,     //OpenJ9-shared_classes_misc
+                                           metadata.getCodeSource(),        //OpenJ9-shared_classes_misc
+                                           metadata.getManifest());         //OpenJ9-shared_classes_misc
```

Comment:
Since we're changing these lines, please correct the indentation (8 spaces more than line 477), but keep the comments aligned.
```java
                        Class<?> clazz = defineClass(name, sharedClazz,     //OpenJ9-shared_classes_misc
                                metadata.getCodeSource(),                   //OpenJ9-shared_classes_misc
                                metadata.getManifest());                    //OpenJ9-shared_classes_misc
```

Reply:
Removed the extra spaces.